### PR TITLE
Add method to accept an already configured Kinesis client

### PR DESCRIPTION
**Pull request by pims**
Based on discussion at: https://github.com/harlow/kinesis-consumer/issues/45

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -15,6 +15,11 @@ func NewKinesisClient() *KinesisClient {
 	return &KinesisClient{svc}
 }
 
+// NewConfiguredKinesisClient returns a configured client to interface with Kinesis stream
+func NewConfiguredKinesisClient(svc *kinesis.Kinesis) *KinesisClient {
```
Comment (harlow) [inline client.go]: can we use functional options on the `NewKinesisClient` function?

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

```go
type Option func(*KinesisClient)

// WithClient overrides the default Kinesis client
func WithClient(svc *kinesis.Kinesis) Option {
	return func(kc *KinesisClient) {
		kc.svc = svc
	}
}

// NewKinesisClient returns client to interface with Kinesis stream
func NewKinesisClient(opts ...Option) *KinesisClient {
	kc := &KinesisClient{}

	for _, opt := range opts {
		opt(kc)
	}

	if kc.svc == nil {
		kc.svc = kinesis.New(session.New(aws.NewConfig()))
	}

	return kc
}
```

that way the `NewKinesisClient` constructor doesn't need to change

The caller would do something like this:

```go
consumer = consumer.NewKinesisClient(consumer.WithClient(kc))

**Comment by harlow** [comment]
For creating a new consumer you can override the client too:

https://github.com/harlow/kinesis-consumer#options

```go
// client
client := kinesis.New(session.New(aws.NewConfig()))

// consumer
c, err := consumer.New(streamName, consumer.WithClient(client))
```

**Comment by pims** [comment]
Rebased it based on your suggestion.

**Comment by harlow** [comment]
thanks for the PR. agree the `client` terminology is a little confusing. there might be some room for improvement there 👍

**Pull request merged**